### PR TITLE
Adding a option to getParams and expandParams

### PR DESCRIPTION
**Pull request by seltix5**
currently the only function to set params ( "setParams" ) will replace the current data with new data, this is not always enough.
Adding "getParams" and "expandParams" will add the option to manipulate the params without erase them.
Without this options it is not possible to update multiple times the params from different sources.

i think this 2 simple functions are useful, what do you think?

### CODE REVIEW

**Review by rnicholus — CHANGES_REQUESTED** [review]
This does seem useful, but in order to merge this in, I’ll need two more updates:

- documentation
- _comprehensive_ unit tests that cover these new methods.

**Comment by seltix5** [comment]
I do can update the documentation, but about the tests I dont know how it works or what I need to do :/

**Comment by seltix5** [comment]
Docs updated. Can you help me with the tests?

**Comment by rnicholus** [comment]
Sure - i'll try to make some time to write these soon.

File: docs/api/methods.jmd [hunk 1]
```diff
@@ -313,6 +313,20 @@ A `resizeInfo` object, which will be passed to the supplied function, contains t
 
 ]) }}
 
+{{ api_method("getParams", "getParams (params[, id])", "Get the parameters for an upload request. Pass in a file id to make the parameters specific to that file.",
+[
+    {
+        "name": "params",
+        "type": "Object",
+        "description": "The parameters to include in the upload request."
```
Comment (SinghSukhdeep) [inline docs/api/methods.jmd]: @seltix5 this is incorrect as `getParams` method doesn't need this parameter. Can you please correct this?.
I'll be happy to write tests for this

**Review by singhjusraj — CHANGES_REQUESTED** [review]

File: docs/api/methods.jmd (hunk 1 above)
Comment (seltix5) [reply docs/api/methods.jmd]: done ^^

File: docs/api/methods.jmd [hunk 2]
```diff
@@ -313,6 +313,20 @@ A `resizeInfo` object, which will be passed to the supplied function, contains t
 
 ]) }}
 
+{{ api_method("getParams", "getParams (id)", "Get the parameters for an upload request. Pass in a file id to make the parameters specific to that file.",
+[
+    {
```
Comment (SinghSukhdeep) [inline docs/api/methods.jmd]: @seltix5 this object should also be removed from here and return statement needs to return an object
Currently you have `null` there, it should be like:

```
[
    {
        "type": "Object",
        "description": "Object containing parameters for the upload request"
    }

]
```

**Pull request closed without merging**